usd/src/usd_parser/USDMaterial.cc

```diff
@@ -39,18 +39,41 @@ namespace sdf
   /// \param[in] _dest The destination for the copy of _ori
   /// \return A list of UsdErrors. An empty list means no errors occurred when
   /// copying _ori to _dest
-  UsdErrors copyFile(const std::string &_ori, const std::string &_dest)
+  UsdErrors copyFile(const std::string &_ori, std::string &_dest)
   {
     UsdErrors errors;
     if (ignition::common::exists(_ori))
     {
-      // If the file exists then we should remove it and we will copy the
-      // file below
-      if (ignition::common::exists(_dest))
+      // If the file exists then we append a number suffix to the destination
+      // file
+      // For example:
+      // /bar/foo.extension
+      // /bar/foo_X.extension
+
+      std::string tmpDest = _dest;
+      std::string parentPath = ignition::common::parentPath(_dest);
+      std::string::size_type fileExtensionIndex = tmpDest.rfind(".");
+
+      std::string fileExtension = "." + tmpDest.substr(fileExtensionIndex + 1);
+      std::string fileNameWithoutExtension = ignition::common::basename(_dest);
+      size_t pos = fileNameWithoutExtension.find(fileExtension);
+      if (pos != std::string::npos)
+      {
+        // If found then erase it from string
+        fileNameWithoutExtension.erase(pos, fileExtension.length());
+      }
+      int index = 0;
+      while (ignition::common::exists(tmpDest))
       {
-        ignition::common::removeFile(_dest);
+        tmpDest = ignition::common::joinPaths(
```

Comment:
I actually don't think the `tmpDest` variable is needed, since the original `_dest` variable is never updated until after the while loop. You could make the loop look like this, and then remove the assignment of `tmpDest` to `_dest` after the while loop since `_dest` is now being updated in the loop as needed (this would allow removal of the `tmpDest` variable completely; we can use `_dest` wherever `tmpDest` is currently being used):
```c++
while (ignition::common::exists(_dest))
{
  _dest = ignition::common::jointPaths(...);
  index++;
}
```